src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -1143,6 +1143,17 @@ private CompoundUseSiteInfo<AssemblySymbol> GetNewCompoundUseSiteInfo()
         /// </summary>
         private sealed class LocalRewritingValidator : BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
         {
+            public override BoundNode? Visit(BoundNode? node)
+            {
+                if (node is BoundIfStatement)
```

Comment:
I don't understand what this changes. What does this get us?

Reply:
> I don't understand what this changes. What does this get us?

It gets us a validation that we don't have ```BoundIfStatement```s post lowering. We no longer can enforce that by overriding ```VisitIfStatement``` because it is now sealed in base. I don't think it makes sense to "unseal" it for the purpose of debug-only visitor.

Reply:
> We no longer can enforce that by overriding VisitIfStatement because it is now sealed in base.

Ah, that's what I was missing. Thanks.